torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -1064,28 +1064,53 @@ def render(  # type: ignore[override]
                 op = self.swap_XW(op)
                 should_swap_xw = True
 
-        if epilogue_nodes:
-            (
-                evt_read_names,
-                evt_write_names,
-                var_name_to_buffer_name,
-                evt_py_code,
-            ) = CutlassEVTCodegen.ir_to_evt_python_code(
-                Y.get_name(), epilogue_nodes, V.kernel.removed_buffers
-            )
-            read_names = OrderedSet(evt_read_names) - OrderedSet(evt_write_names)
-            write_names = OrderedSet(evt_write_names)
-            assert write_names, "There should be at least one write"
-            D_output_name = var_name_to_buffer_name["D"]
-            name_to_buffer = V.graph.name_to_buffer | V.graph.graph_inputs
-            D_output_buffer = name_to_buffer[D_output_name]
-            Y = D_output_buffer  # type: ignore[assignment]
-            # Interestingly, I don't think the rest of the layout matters here since we
-            # use the properties of the Y buffer to fill in D's properties in the epilogue
-            # args. This is needed though because it defines types expected in the epilogue args.
-            op.D.element = cutlass_utils.torch_dtype_to_cutlass_type(
-                D_output_buffer.get_layout().dtype
-            )
+        if epilogue_nodes or len(self.input_nodes) == 4:
```

Comment:
can we make this a bit more explicit ? theres nothing from reading the code locally that would tell you what len(self.input_nodes) == 4 means.

Reply:
ended up naming this since I had to merge anyway